be/src/exec/vectorized/hdfs_scanner.h

```diff
@@ -43,6 +45,28 @@ struct HdfsScanStats {
     int64_t group_dict_decode_ns = 0;
 };
 
+class HdfsParquetProfile;
```

Comment:
Should we extract a base class `HdfsScannerProfile` letting `HdfsParquetProfile` to be derived from it. create a virtual method `HdfsScanner::create_profile` letting each derived class to create their own profile. Finally, `HdfsScanNode` doesn't have to know anything about `parquet` or `ORC`

Reply:
It's quite hard.  Beause profile is a part of hdfs scan node, and in theory a hdfs scan node can have both parquet and orc scanner. But each scanner can update only its own profile part.